src/Sylius/Bundle/UiBundle/Resources/translations/messages.en.yml

```diff
@@ -645,6 +645,7 @@ sylius:
         product_associations: 'Product associations'
         product_attributes: 'Product attributes'
         product_details_viewing_product_product: 'Product details <small>Viewing product "<strong>%product%</strong>"</small>'
+        product_name: 'Product name'
```

Comment:
Do you need it? on Order show page similar column is named as "item" IMO here just "Item" or "Product" should be enough 
same situation with variant below.